### PR TITLE
Commiting 2 functions after updating the suggestions by reviewers 

### PR DESCRIPTION
**Pull request by ShyamBhuptani**
Hi @flexplicateur,

I have updated the code as per your suggestion and reapplied this pull request.
The previous pull request is closed but all comments are thoroughly understood and re-implemented.
This request mainly contains 2 functionalities.
1. Support Degree Matrix 
2. Contribution

I have also applied the changes you have suggested.
1. Using fabs function ( I am working in ubuntu so, to mitigate -lm while compilation for math.h, i used ternary, but still I have replaced it with fabs)
2. Argument sequence for function array then integers
3. Spacing in for loop has been corrected
4. More dense comments are provided for main_test.c
5. Variable name has changed for temperature 

For the first point you mentioned in your comments related to all functioned designed in separate C files was decided by me and Amuleen as we wanted to build something which is reusable. Though it is lengthy to make all functions separate, it can be useful for any other identical projects for various scenarios.

### CODE REVIEW

**Comment by fcharih** [comment]
There is no benefit to having all the functions in separate files. It doesn't really improve reusability. In fact, the user would need to import multiple files to access the needed functions instead of a single one. I will not enforce this, but this is not a good practice.

File: SensorFusion_src/main_test.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#include "../include/support_degree_matrix.h"
+#include "../include/contribution.h"
+#define num1 2
+#define num2 3
+
+// Main test function
+int main(int argc, char ** argv){
+
+    /*Testing of Function Support Degree Matrix using one-D input array
+    * This fuction performs (e^-(abolute(reading[i]-reading[j]))) for 2 sensors
```
Comment (fcharih) [inline SensorFusion_src/main_test.c]: typo: function

File: SensorFusion_src/main_test.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* Author : Shyam Bhuptani 
+* Function : Unit Testing of all Functions
+*/
+
+/*Including all dependancies*/
+#include <stdio.h>
+#include <stdlib.h>
+#include "../include/support_degree_matrix.h"
+#include "../include/contribution.h"
+#define num1 2
```
Comment (fcharih) [inline SensorFusion_src/main_test.c]: Not a good variable name. num1 doesn't mean anything to me.

File: SensorFusion_src/contribution.c [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+*/
+
+/*Including dependancies*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include "../include/contribution.h" 
+
+double * contribution(double lambda[], int n)
```
Comment (fcharih) [inline SensorFusion_src/contribution.c]: You should make your function name more explicit. Ideally, the function name should describe what the function does, eg. compute_contributions(). It is unclear how the function relates to contributions, in this case.

File: SensorFusion_src/suppoort_degree_matrix.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+*/
+
+/*Including dependancies*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "../include/support_degree_matrix.h"
+
+/*Function declaration which returns a 2D array */
+double ** support_degree_matrix(double sensor_readings[], int n)
```
Comment (fcharih) [inline SensorFusion_src/suppoort_degree_matrix.c]: Same idea here. The function should describe the action of the function, e.g. **build**_sdm().

File: include/Support_Degree_Matrix.h [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#ifndef Support_Degree_Matrix_h
```
Comment (fcharih) [inline include/Support_Degree_Matrix.h:1]: Should follow the convention described in CULearn for header guards.

File: include/contribution.h [hunk 6]
```diff
@@ -1 +1,13 @@
-//header file for contribution.c source file
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (fcharih) [inline include/contribution.h]: When you merge, you should pick one of the two versions in a conflict. Besides, the header guards do not follow the convention. See convention list on CULearn.

**Review by fcharih — CHANGES_REQUESTED** [review]
Careful, your code does not comply with the coding conventions described on CULearn:

1.  Function names are not very descriptive, they should describe the action that the function does.
2.  Header guards do not follow the correct convention
3.  Inconsistent indentation and spacing. An indent should be 4 spaces.

I strongly advise you to go have a look at the coding convention and to ensure that all the rules are respected in your code.

File: SensorFusion_src/main_test.c (hunk 1 above)
Comment (ShyamBhuptani) [reply SensorFusion_src/main_test.c]: Updated in new commit, Please check

File: SensorFusion_src/main_test.c (hunk 2 above)
Comment (ShyamBhuptani) [reply SensorFusion_src/main_test.c]: Updated in new commit, Please check

File: SensorFusion_src/contribution.c (hunk 3 above)
Comment (ShyamBhuptani) [reply SensorFusion_src/contribution.c]: Updated in new commit, Please check

File: SensorFusion_src/suppoort_degree_matrix.c (hunk 4 above)
Comment (ShyamBhuptani) [reply SensorFusion_src/suppoort_degree_matrix.c]: Updated new name in recent commit, please review

File: include/contribution.h (hunk 6 above)
Comment (ShyamBhuptani) [reply include/contribution.h]: <<<<<HEAD was inserted while working on VS code.
They should be erased in new commit

**Review by fcharih — APPROVED** [review]
OK

**Review by amuleengulati — APPROVED** [review]

**Pull request merged**